consensus/propagation/types.go

```diff
@@ -0,0 +1,230 @@
+package types
+
+import (
+	"errors"
+
+	"github.com/tendermint/tendermint/crypto/merkle"
+	"github.com/tendermint/tendermint/libs/bits"
+	protoprop "github.com/tendermint/tendermint/proto/tendermint/propagation"
+	"github.com/tendermint/tendermint/types"
+)
+
+// TxmetaData keeps track of the hash of a transaction and its location with the
+// protobuf encoded block.
+type TxMetaData struct {
+	Hash  []byte `protobuf:"bytes,1,opt,name=hash,proto3" json:"hash,omitempty"`
+	Start uint32
+	End   uint32
```

Comment:
[nit]
would be nice to specify how inclusive/exclusive there ranges are. Maybe just a comment saying that the range is [start, End)